src/screens/crew6/scienceScreen.cpp

```diff
@@ -203,18 +219,123 @@ ScienceScreen::ScienceScreen(GuiContainer* owner, ECrewPosition crew_position)
+
+    // Biological view toggle.
+    signal_details_biological_button = new GuiToggleButton(this, "SIGNAL_DETAILS_BIOLOGICAL", "B", [this](bool value){
+        setBiologicalDetailsToggle(value);
+    });
+    signal_details_biological_button->setValue(false)->setPosition(-270, -70, ABottomRight)->setSize(50, 50)->setVisible(false);
+    signal_details_biological_button->setColors(colorConfig.button_blue);
+
+    // Radar/database view toggle.
+    view_mode_selection = new GuiListbox(this, "VIEW_SELECTION", [this](int index, string value) { setViewModeSelectionToggle(index); });
     view_mode_selection->setOptions({"Radar", "Database"})->setSelectionIndex(0)->setPosition(20, -20, ABottomLeft)->setSize(200, 100);
 
     // Scanning dialog.
     new GuiScanningDialog(this, "SCANNING_DIALOG");
 }
 
+void ScienceScreen::setViewModeSelectionToggle(int index)
+{
+    radar_view->setVisible(index == 0);
+    background_gradient->setVisible(index == 0);
+    database_view->setVisible(index == 1);
+
+    // Switching to database view resets signal details if available.
+    if (my_spaceship->has_signal_radar)
+    {
+        signal_details_toggle->setValue(false)->setVisible(index == 0);
+        setSignalDetailsToggle(false);
+        signal_details_visual_button->setVisible(false);
+        signal_details_electrical_button->setVisible(false);
+        signal_details_gravity_button->setVisible(false);
+        signal_details_biological_button->setVisible(false);
+    }
+}
+
+void ScienceScreen::setSignalDetailsToggle(bool value)
+{
+    if (my_spaceship->has_signal_radar)
+    {
+        // Toggle visibility of signal lens toggles.
+        signal_details_visual_button->setValue(true)->setVisible(value);
+        signal_details_electrical_button->setValue(false)->setVisible(value);
+        signal_details_gravity_button->setValue(false)->setVisible(value);
+        signal_details_biological_button->setValue(false)->setVisible(value);
+
+        // Toggle and reset signal details.
+        science_radar->setSignalDetails(value);
+        probe_radar->setSignalDetails(value);
+        science_radar->setVisualObjects(true)->setSignalElectrical(false)->setSignalGravity(false)->setSignalBiological(false);
+        probe_radar->setVisualObjects(true)->setSignalElectrical(false)->setSignalGravity(false)->setSignalBiological(false);
+    }
+}
+
+void ScienceScreen::setVisualDetailsToggle(bool value)
+{
+    if (my_spaceship->has_signal_radar && science_radar->getSignalDetails())
+    {
+        science_radar->setVisualObjects(value)->setSignalElectrical(false)->setSignalGravity(false)->setSignalBiological(false);
+        probe_radar->setVisualObjects(value)->setSignalElectrical(false)->setSignalGravity(false)->setSignalBiological(false);
+        signal_details_electrical_button->setValue(false);
+        signal_details_gravity_button->setValue(false);
+        signal_details_biological_button->setValue(false);
```

Comment:
I would suggest that you change
`        science_radar->setVisualObjects(value)->setSignalElectrical(false)->setSignalGravity(false)->setSignalBiological(false);
        probe_radar->setVisualObjects(value)->setSignalElectrical(false)->setSignalGravity(false)->setSignalBiological(false);
        signal_details_electrical_button->setValue(false);
        signal_details_gravity_button->setValue(false);
        signal_details_biological_button->setValue(false);`
To
`        science_radar->setVisualObjects(value)->setSignalElectrical(!value)->setSignalGravity(!value)->setSignalBiological(!value);
        probe_radar->setVisualObjects(value)->setSignalElectrical(!value)->setSignalGravity(!value)->setSignalBiological(!value);
        signal_details_electrical_button->setValue(!value);
        signal_details_gravity_button->setValue(!value);
        signal_details_biological_button->setValue(!value);`
This makes it so when you deselect V it will just select all 3 other buttons. While it still is possible to deselect all 3 sensors and see nothing, it provides a quick way to switch between seeing Visual and seeing all available sensor data.
https://gyazo.com/a68b277506a56e83d9baba71a31c0044